docs/csharp/whats-new/csharp-6.md

```diff
@@ -98,483 +36,157 @@ public class Student
 
 [!code-csharp[AwaitFinally](../../../samples/snippets/csharp/new-in-6/NetworkClient.cs#AwaitFinally)]
 
-The implementation details for adding `await` support inside `catch`
-and `finally` clauses ensures that the behavior is consistent with the
-behavior for synchronous code. When code executed in a `catch` or `finally`
-clause throws, execution looks for a suitable `catch` clause in the next
-surrounding block. If there was a current exception, that exception is
-lost. The same happens with awaited expressions in `catch` and `finally`
-clauses: a suitable `catch` is searched for, and the current exception,
-if any, is lost.  
+The implementation details for adding `await` support inside `catch` and `finally` clauses ensure that the behavior is consistent with the behavior for synchronous code. When code executed in a `catch` or `finally` clause throws, execution looks for a suitable `catch` clause in the next surrounding block. If there was a current exception, that exception is lost. The same happens with awaited expressions in `catch` and `finally` clauses: a suitable `catch` is searched for, and the current exception, if any, is lost.  
 
 > [!NOTE]
-> This behavior is the reason it's recommended to write `catch` and `finally`
-> clauses carefully, to avoid introducing new exceptions.
+> This behavior is the reason it's recommended to write `catch` and `finally` clauses carefully, to avoid introducing new exceptions.
 
-## Index initializers
+## Initialize associative collections using indexers
 
-*Index Initializers* is one of two features that make collection
-initializers more consistent with index usage. In earlier releases of C#, you could use
-*collection initializers* only with sequence style collections, including <xref:System.Collections.Generic.Dictionary%602> by adding braces around key and value pairs:
+*Index Initializers* is one of two features that make collection initializers more consistent with index usage. In earlier releases of C#, you could use *collection initializers* with sequence style collections, including <xref:System.Collections.Generic.Dictionary%602> by adding braces around key and value pairs:
 
-[!code-csharp[ListInitializer](../../../samples/snippets/csharp/new-in-6/initializers.cs#ListInitializer)]
+[!code-csharp[ListInitializer](../../../samples/snippets/csharp/new-in-6/initializers.cs#CollectionInitializer)]
 
-Now, you can use them with <xref:System.Collections.Generic.Dictionary%602> collections and similar types. The new syntax supports assignment using an index into the collection:
+Now, you can use them with <xref:System.Collections.Generic.Dictionary%602> collections and other types where the accessible `Add` method accepts more than one argument. The new syntax supports assignment using an index into the collection:
```

Comment:
The rationale behind this suggestion is in the [comment](https://github.com/dotnet/docs/pull/9551#discussion_r245432130)

```suggestion
You can use them with <xref:System.Collections.Generic.Dictionary%602> collections and other types where the accessible `Add` method accepts more than one argument. The new syntax supports assignment using an index into the collection:
```